libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -111,3 +125,42 @@ fun createNavigationRoutes(
 fun RouteInterface.toNavigationRoute(): NavigationRoute {
     return this.toNavigationRoute()
 }
+
+private fun List<LegStep>.updateSteps(
+    route: DirectionsRoute,
+    mergedAnnotation: LegAnnotation?
+): List<LegStep> {
+    val mergedDurations = mergedAnnotation?.duration() ?: return this
+    val result = mutableListOf<LegStep>()
+    var previousStepsAnnotationsCount = 0
+    for (step in this) {
+        val stepPointsSize = route.stepGeometryToPoints(step).size
```

Comment:
@LukasPaczos , can you please confirm that Navigation SDK doesn't support geojson as geomerties?

Reply:
as I see we don't even have this option in mapbox-java despite the fact that it's present on Directions API

Reply:
That's correct. Historically, we didn't add geojson support because the responses were significantly bigger when that type of geometry was used. Given how response has been growing, it might now be a smaller percentage increase, but we have yet to measure that.